test/test_base_metric.py

```diff
@@ -5,86 +5,108 @@
 from sklearn import clone
 from sklearn.utils.testing import set_random_state
 from test.test_utils import ids_metric_learners, metric_learners
+import string
+
+
+def remove_spaces(s):
+  s = s.replace(' ', '')
+  s = s.replace('\n', '')
+  return s
 
 
 class TestStringRepr(unittest.TestCase):
 
   def test_covariance(self):
-    self.assertEqual(str(metric_learn.Covariance()),
-                     "Covariance(preprocessor=None)")
+    self.assertEqual(remove_spaces(str(metric_learn.Covariance())),
+                     remove_spaces("Covariance(preprocessor=None)"))
 
   def test_lmnn(self):
     self.assertRegexpMatches(
         str(metric_learn.LMNN()),
         r"(python_)?LMNN\(convergence_tol=0.001, k=3, learn_rate=1e-07, "
-        r"max_iter=1000,\n      min_iter=50, preprocessor=None, "
-        r"regularization=0.5, use_pca=True,\n      verbose=False\)")
+        r"max_iter=1000,\n(\ +)min_iter=50, preprocessor=None, "
+        r"regularization=0.5, use_pca=True,\n(\ +)verbose=False\)")
 
   def test_nca(self):
-    self.assertEqual(str(metric_learn.NCA()),
-                     "NCA(max_iter=100, num_dims=None, preprocessor=None, "
-                     "tol=None, verbose=False)")
+    self.assertEqual(remove_spaces(str(metric_learn.NCA())),
+                     remove_spaces(
+                       "NCA(max_iter=100, num_dims=None, preprocessor=None, "
+                       "tol=None, verbose=False)"))
 
   def test_lfda(self):
-    self.assertEqual(str(metric_learn.LFDA()),
-                     "LFDA(embedding_type='weighted', k=None, num_dims=None, "
-                     "preprocessor=None)")
+    self.assertEqual(remove_spaces(str(metric_learn.LFDA())),
+                     remove_spaces(
+                       "LFDA(embedding_type='weighted', k=None, "
+                       "num_dims=None, "
+                       "preprocessor=None)"))
 
   def test_itml(self):
-    self.assertEqual(str(metric_learn.ITML()), """
+    self.assertEqual(remove_spaces(str(metric_learn.ITML())),
+                     remove_spaces("""
 ITML(A0=None, convergence_threshold=0.001, gamma=1.0, max_iter=1000,
    preprocessor=None, verbose=False)
-""".strip('\n'))
-    self.assertEqual(str(metric_learn.ITML_Supervised()), """
+""".strip('\n')))
+    self.assertEqual(remove_spaces(str(metric_learn.ITML_Supervised())),
+                     remove_spaces("""
 ITML_Supervised(A0=None, bounds='deprecated', convergence_threshold=0.001,
         gamma=1.0, max_iter=1000, num_constraints=None,
         num_labeled='deprecated', preprocessor=None, verbose=False)
-""".strip('\n'))
+""".strip('\n')))
```

Comment:
No need to strip newlines if you're also calling `remove_spaces()`, right?

Reply:
That's right indeed